resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java

```diff
@@ -61,7 +63,9 @@ public Object bulkheadAroundAdvice(ProceedingJoinPoint proceedingJoinPoint, Bulk
 		}
 		String backend = backendMonitored.name();
 		io.github.resilience4j.bulkhead.Bulkhead bulkhead = getOrCreateBulkhead(methodName, backend);
-		return handleJoinPoint(proceedingJoinPoint, bulkhead, methodName);
+		RecoveryFunction recovery = RecoveryFunctionUtils.getInstance(backendMonitored.recovery());
```

Comment:
Can we move the creation of a RecoveryFunction into the `handleJoinPoint` method and only create an instance when an exception is catched?

Reply:
That is a good idea. I moved it to the catch block.